src/components/Calendar/ClassModal.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/styles';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import Typography from '@material-ui/core/Typography';
+import Paper from '@material-ui/core/Paper';
+import Button from '@material-ui/core/Button';
+
+export const styles = {
+    modal: {
+        position: 'absolute',
+        width: '450px',
+        outline: 'none',
+        top: '12.5%',
+        left: '12.5%',
+        padding: '10px',
+    },
+};
+
+class ClassModal extends Component {
+    constructor (props) {
+        super(props);
+        this.state = {
+            showDialog: props.showDialog,
```

Comment:
Here, you're currently initializing this component's internal state based on the props it gets from its parent. This is a common anti-pattern that you'll want to try to avoid (there are rare cases when this is necessary, but this isn't one of them). This is an anti-pattern because if you initialize state using props, when that prop later changes, the state won't.

Generally in React, it's a really good idea to keep components stateless if you can. Think of it this way: In this case, all you want is a modal that displays some information that's passed into the modal component as props. This modal component doesn't care whether it's open or closed - its parent can take care of showing or not showing it. So this component can just be a functional stateless component! Much simpler.